src/node/http.ts

```diff
@@ -927,7 +927,29 @@ export class HttpServer {
     }
 
     // Must be authenticated to use the proxy.
-    route.provider.ensureAuthenticated(request)
+    if (!route.provider.authenticated(request)) {
+      // Attempt to determine if it's the user browsing the root and if so fall
+      // through to allow the login flow.
+      if (request.headers["content-type"] !== "application/json") {
```

Comment:
Let's check explicitly for `text/html` here.

Reply:
I wanted to do that but Firefox and Chromium don't seem to set this at all. When I logged it the value was `undefined`.